// @groot/falbala/fractal.js
'use strict'

const path = require('path')
const fractal = module.exports = require('@frctl/fractal').create()
const pkg = require(path.join(__dirname, 'package.json'))

fractal.set('project.title', 'Falbala')
fractal.set('project.version', pkg.version)

fractal.components.set('path', path.join(__dirname, 'components'))
fractal.components.set('ext', '.html')
fractal.components.set('label', 'Patterns')

fractal.docs.set('path', path.join(__dirname, 'docs'))

fractal.web.set('static.path', path.join(__dirname, 'tmp'))
fractal.web.set('server.sync', true)
fractal.web.set('builder.dest', path.join(__dirname, 'build'))

fractal.components.set('statuses', {
  prototype: {
    label: 'Prototype',
    description: 'Do not implement.',
    color: '#FF3333'
  },
  wip: {
    label: 'In Progress',
    description: 'Work in progress. Implement with caution.',
    color: '#ffa557'
  },
  ready: {
    label: 'Ready',
    description: 'Ready to implement.',
    color: '#29CC29'
  }
})

const dreiTheme = require('./theme')
fractal.web.theme(dreiTheme)

/**
 * Prevent Bluebird warnings like "a promise was created in a handler but was not returned from it"
 * caused by Nunjucks from polluting the console
 */
const bluebird = require('bluebird')
bluebird.config({
  warnings: false
})
